eng/common/TestResources/New-TestResources.ps1

```diff
@@ -537,19 +569,21 @@ try {
                 $displayName = "$($baseName)$suffix.test-resources.azure.sdk"
             }
 
+
             $servicePrincipal = Retry {
                 New-AzADServicePrincipal -Role "Owner" -Scope "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName" -DisplayName $displayName
             }
+            $servicePrincipalWrapper = NewServicePrincipalWrapper $servicePrincipal
```

Comment:
Should we consider bundling the `New-AzADServicePrincipal` call and password creation into a `CreateServicePrincipal` function?   It's not obvious that we've got an implicit "create the password" inside of the wrapper; both steps seem to logically be part of the principal creation operation.